CMakeLists.txt

```diff
@@ -14,16 +14,63 @@ find_package(Python 3.9 COMPONENTS Interpreter Development.Module REQUIRED ${SKB
 find_package(nanobind CONFIG REQUIRED)
 
 option(NIFTY_LS_OPENMP "Enable OpenMP support in the compiled extensions" ON)
+option(NIFTY_LS_HETEROBATCH "Enable heterobatch support with finufft" On)
 
 if(NIFTY_LS_OPENMP)
     find_package(OpenMP REQUIRED)
 endif()
 
-set(NIFTY_LS_MODULES cpu_helpers chi2_helpers)
-
 # until https://github.com/wjakob/nanobind/pull/1092 is published
 add_compile_options(-Wno-deprecated-literal-operator)
 
+if(NIFTY_LS_HETEROBATCH)
+    # Download CPM.cmake if not already available
+    if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/cmake/CPM.cmake)
+        file(DOWNLOAD https://github.com/cpm-cmake/CPM.cmake/releases/download/v0.42.0/CPM.cmake
+             ${CMAKE_CURRENT_BINARY_DIR}/cmake/CPM.cmake)
+    endif()
+    include(${CMAKE_CURRENT_BINARY_DIR}/cmake/CPM.cmake)
+    
+    # Add finufft
+    CPMAddPackage(
+        NAME finufft
+        GITHUB_REPOSITORY flatironinstitute/finufft
+        GIT_TAG v2.4.1
+        OPTIONS
+            "FINUFFT_USE_OPENMP OFF"
+            "FINUFFT_USE_CPU ON"
+            "FINUFFT_STATIC_LINKING ON"
+            "FINUFFT_POSITION_INDEPENDENT_CODE ON"
+            # "FINUFFT_ARCH_FLAGS -march=native"
+    )
+
+    set(NIFTY_LS_HETEROBATCH_MODULES finufft_heterobatch_helpers finufft_chi2_heterobatch_helpers)
+
+    foreach(MODULE_NAME ${NIFTY_LS_HETEROBATCH_MODULES})
```

Comment:
Do we need to repeat this CMake logic or could we just add the heterobatch modules to `NIFTY_LS_MODULES` and use the existing loop below? They also still need to be linked against finufft, but I think that's the only specialization.